analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantNullCheck.cs

```diff
@@ -155,20 +155,15 @@ private void CheckOrPattern(SyntaxNodeAnalysisContext context)
             {
                 var leftPattern = (PatternSyntaxWrapper)left;
                 var rightPattern = (PatternSyntaxWrapper)right;
-                if (leftPattern.IsNull() && IsNegativePatternMatch(rightPattern))
+                if (leftPattern.IsNull() && rightPattern.IsNot())
```

Comment:
I'm pretty sure I am missing something in the test cases, but they are green and the code coverage was incomplete, thus that previous logic was not necessary